mixer/adapter/stackdriver/metric/bufferedClient.go

```diff
@@ -99,15 +119,91 @@ func (b *buffered) Send() {
 		// We need to build framework level support for these kinds of async tasks. Perhaps a generic batching adapter
 		// can handle some of this complexity?
 		if err != nil {
+			ets := handleError(err, timeSeries)
+			b.updateRetryBuffer(ets)
 			_ = b.l.Errorf("Stackdriver returned: %v\nGiven data: %v", err, timeSeries)
 		} else {
 			b.l.Debugf("Successfully sent data to Stackdriver.")
 		}
 	}
 }
 
+// Retry all time series in retry buffer. If still fails, add the failed time series to the normal
+// buffer so that it will be merged with newer timeseries. Otherwise if a newer timeseries was
+// successfully written, it can never be written.
+func (b *buffered) retry() {
+	toRetry := make([]*monitoringpb.TimeSeries, 0, len(b.retryBuffer))
+	keyMap := make(map[*monitoringpb.TimeSeries]uint64)
+	for k, r := range b.retryBuffer {
+		toRetry = append(toRetry, r.ts)
+		keyMap[r.ts] = k
+	}
+	retryBatches := batchTimeSeries(toRetry, b.timeSeriesBatchSize)
+	for _, timeSeries := range retryBatches {
+		err := b.pushMetrics(context.Background(),
+			&monitoringpb.CreateTimeSeriesRequest{
+				Name:       "projects/" + b.project,
+				TimeSeries: timeSeries,
+			})
+		if err != nil {
+			ets := handleError(err, timeSeries)
+			b.Record(ets)
+		} else {
+			for _, ts := range timeSeries {
+				delete(b.retryBuffer, keyMap[ts])
+			}
+		}
+	}
+}
+
 func (b *buffered) Close() error {
 	b.l.Infof("Sending last data before shutting down")
 	b.Send()
 	return b.closeMe.Close()
 }
+
+// handleError extract out timeseries that fails to create from response status.
+// If no sepecific timeseries listed in error response, retry all time series in batch.
+func handleError(err error, tsSent []*monitoringpb.TimeSeries) []*monitoringpb.TimeSeries {
+	errorTS := make([]*monitoringpb.TimeSeries, 0, 0)
+	retryAll := true
+	if s, ok := status.FromError(err); ok {
+		sd := s.Details()
+		for _, i := range sd {
+			if t, ok := i.(*monitoringpb.CreateTimeSeriesError); ok {
+				retryAll = false
+				if isOutOfOrderWrite(t.GetStatus()) {
+					// Don't deal with out of order error: the point can never be written.
+					continue
+				}
+				errorTS = append(errorTS, t.GetTimeSeries())
+			}
+		}
+	}
+	if retryAll {
```

Comment:
You should check the overall error status against isRetryable as well.

Reply:
Done.